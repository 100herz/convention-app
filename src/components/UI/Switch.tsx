import React from 'react'
import { StyleSheet, Switch as DefaultSwitch, ViewStyle } from 'react-native'

import { colors, DefaultStyles, defaultStyles } from '@styles/theme'

interface Props {
  style?: ViewStyle
  testID?: string
  onValueChange: (value: boolean) => void
  value: boolean
}

const Switch: React.FC<Props> = ({ style, onValueChange, value, testID }) => {
  return (
    <DefaultSwitch
      style={{ ...styles.switch, ...style }}
      trackColor={{ false: colors.gray, true: colors.accentColor }}
      thumbColor={colors.grayLight}
      ios_backgroundColor={colors.gray}
      onValueChange={onValueChange}
      value={value}
      testID={testID}
    />
  )
}

type Styles = DefaultStyles

const styles = StyleSheet.create<Styles>({
  ...defaultStyles,
})

export default Switch
