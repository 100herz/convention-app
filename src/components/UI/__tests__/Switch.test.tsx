/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react'
import { render } from '@testing-library/react-native'

import Switch from '../Switch'

describe('<Switch />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<Switch onValueChange={() => {}} value={true} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
