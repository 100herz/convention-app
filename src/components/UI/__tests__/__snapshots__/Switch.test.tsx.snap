// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text>
    true
  </Text>
</View>
`;
